goto/commands/rename.py

```diff
@@ -0,0 +1,26 @@
+from ..gotomagic.text import GotoWarning
+
+
+def rename(magic, command, args):
+    if len(args) == 0:
+        return None, GotoWarning("missing_both_magicwords", command=command)
+
+    from_magicword = args[0]
+
+    if len(args) == 1:
+        return None, GotoWarning("missing_to_magicword", command=command, magicword=from_magicword)
+
+    to_magicword = args[1]
+
+    if from_magicword not in magic.magic:
+        return None, GotoWarning("magicword_does_not_exist", magicword=from_magicword)
+
+    if to_magicword in magic.magic:
+        return None, GotoWarning("adding_existing_magicword_short", magicword=to_magicword, uri=magic[to_uri])
+
+    from_uri = magic[from_magicword]
```

Comment:
I think the handling of the renaming should be done inside the magic class. 
This is in order to keep a consitent separation of concern - and thererfore the possibilty to interchange the underlaying implementation.

The rename operation should therefore be moved to a method in the GotoMagic class. `rename` or something

Reply:
I totally agree